plugins/obs-qsv11/QSV_Encoder.cpp

```diff
@@ -175,12 +191,22 @@ void qsv_encoder_version(unsigned short *major, unsigned short *minor)
 
 qsv_t *qsv_encoder_open(qsv_param_t *pParams)
 {
+	qsv_adapter_info *adapter_info = new qsv_adapter_info;
+
+	qsv_query_encode_device(adapter_info);
+
 	mfxIMPL impl_list[4] = {MFX_IMPL_HARDWARE, MFX_IMPL_HARDWARE2,
 				MFX_IMPL_HARDWARE3, MFX_IMPL_HARDWARE4};
-	int igpu_index = -1;
-	if (prefer_current_or_igpu_enc(&igpu_index) &&
-	    (igpu_index < _countof(impl_list))) {
-		impl = impl_list[igpu_index];
+
+	//select current adapter - will be iGPU if exists due to adapter reordering
+	impl = impl_list[adapter_info->currentGPUIndex];
+
+	//check iGPU support for AV1 first, if dGPU exists check support for AV1
+	if (pParams->codec == MFX_CODEC_AV1) {
+		bool av1_supported = qsv_av1_supported(impl);
```

Comment:
This appears to check if AV1 encoding is supported every time an encoder is started, I assume due to the possibility that the user has selected a GPU that doesn't support it.

Since this check already happens in `obs_module_load`, would it perhaps be possible to cache this information and simply look it up here instead?

Reply:
As a follow-up, is there perhaps a more graceful way to check capabilities than creating an encoder?

Reply:
We have cleaned up this portion of the code. Please take a look and let us know any other feedback.

Reply:
Yeah this seems good, I don't have the hardware to check functionality so that'll be up to @RytoEX 

There's still the question of what happens with multiple dGPUs, but I'm fine leaving that for later since those setups are exceedingly rare anyway.

Reply:
Will re-test as soon as I can.

Reply:
> Yeah this seems good, I don't have the hardware to check functionality so that'll be up to @RytoEX
> 
> There's still the question of what happens with multiple dGPUs, but I'm fine leaving that for later since those setups are exceedingly rare anyway.

I'm using two GPUs on a single PC and the QSV encoder doesn't start up if OBS is running on a non-Intel GPU (like NVIDIA/AMD). It can be done on the stable OBS build QSV encoder, however.

Reply:
Having the ability to use av1 with another vendor gpu in the system is a must it’s why these gpus are being bought so the question is can this be done?